dataprofiler/data_readers/json_data.py

```diff
@@ -63,21 +64,93 @@ def __init__(self, input_file_path=None, data=None, options=None):
     def selected_keys(self):
         return self._selected_keys
 
+    def _find_data(self, json_lines, path=""):
+        """
+        Finds all the column headers/data in a Json and returns them as a 
+        list.
+
+        :param json_lines: json format list of dicts or dict
+        :type json_lines: list or string or dict
+        :return: list of dicts of {column headers: values}
+        """
+        if path != "":
+            path = path + "."
+
+        list_of_dict = []
+        if isinstance(json_lines, dict):
+            for key in json_lines:
+                if isinstance(json_lines[key], dict): # If nested dictionary
+                    list_of_dict = list_of_dict + self._find_data(
+                        json_lines[key], path + key)
+                elif isinstance(json_lines[key], list): # If list
+                    if isinstance(json_lines[key][0], dict): # If list of dictionaries
+                        for item in json_lines[key]:
+                            list_of_dict = list_of_dict + self._find_data(item, path+key)
+                    else: # if non-dict list
+                        list_of_dict = list_of_dict + [{path+key: json_lines[key]}]
+                else: # If just a dictionary of non-dict non-list values
+                    list_of_dict = list_of_dict + [{path+key: json_lines[key]}]
+        else: 
+            list_of_dict = list_of_dict + [{path: json_lines}]
+        return list_of_dict
+
+
     def _load_data_from_json_format(self, json_lines):
         """
         Loads the data when in a JSON format such as
         `json_lines = [dict(a=1), dict(a=2)]`
-        
+
         :param json_lines: json format list of dicts or dict
         :return:
         """
-        if isinstance(json_lines, dict):
-            json_lines = [json_lines]
-        data, original_df_dtypes = data_utils.json_to_dataframe(
-            json_lines=json_lines,
-            selected_columns=self.selected_keys,
-            read_in_string=False
-        )
+        if self._intuitive_detection:
+            bulk_data = None
+            if isinstance(json_lines, dict):
+                if "data" in json_lines.keys():
```

Comment:
We might want to add a comment here or check several terms (`payload` for instance)